src/main/java/PasswordGenerator.java

```diff
@@ -22,18 +21,24 @@ public String createStringOfRandomCharacters() {
 
         String randomCharacters = "";
 
+        //This could be a for loop
         randomCharacters += pickCharactersFromList(1, characterTypeList.get(0));
         randomCharacters += pickCharactersFromList(1, characterTypeList.get(1));
         randomCharacters += pickCharactersFromList(1, characterTypeList.get(2));
 
         return randomCharacters;
     }
 
+    //method visibility?
     public String pickCharactersFromList(int numberAmount, String alphabet) {
+        //variable naming?
         int N = alphabet.length();
+        //maybe character isn't the best name?
         String character = "";
+        //resuse the Random instance?
         Random r = new Random();
```

Comment:
You should be able to reuse the same Random object rather than creating a new one every time, might have a bit of a performance saving.